src/main/kotlin/calculator/Calculator.kt

```diff
@@ -0,0 +1,33 @@
+package calculator
+
+class Calculator {
+    fun calculate(expression: String?): Int {
+        validate(expression)
+
+        val elements = expression!!.split(" ").groupBy { isNumeric(it) }
```

Comment:
코틀린에서는 컴파일러가 수행된 타입 검사에 따라 자동으로 캐스팅해 주며 이를 스마트 캐스트(smart cast)라고 합니다. 그러나 이러한 검사를 별도의 함수로 분리하면 스마트 캐스트가 사라집니다. 계속 `expression`을 사용한다면 캐스팅하는 것이 편리할 수 있어요.

```suggestion
        if (expression.isNullOrEmpty()) {
            throw IllegalArgumentException("빈문자열은 입력될 수 없습니다.")
        }

        val elements = expression.split(" ").groupBy { isNumeric(it) }
```